librz/core/cmd/cmd_print.c

```diff
@@ -1716,48 +1716,83 @@ static void pr_bb(RzCore *core, RzAnalysisFunction *fcn, RzAnalysisBlock *b, boo
 	}
 }
 
-static void disasm_until_ret(RzCore *core, ut64 addr, int limit, RzOutputMode mode) {
-	int p = 0;
+static void handle_default_disasm_print_mode(const RzCore *core, const ut64 addr, const bool ret_val,
+	const char *m_intr, RzStrBuf *buf) {
 	const bool show_color = rz_config_get_i(core->config, "scr.color");
+
+	if (show_color) {
+		const char *offsetColor = rz_cons_singleton()->context->pal.offset; // TODO etooslow. must cache
+		if (!ret_val) {
+			rz_cons_printf("%s0x%08" PFMT64x "" Color_RESET "  %10s %s\n",
+				offsetColor, addr, "", m_intr);
+		} else {
+			char *str_buf = rz_str_newf("%s0x%08" PFMT64x "" Color_RESET "  %10s %s\n",
+				offsetColor, addr, "", m_intr);
+			rz_strbuf_append(buf, str_buf);
+			free(str_buf);
+		}
+	} else {
+		if (!ret_val) {
+			rz_cons_printf("0x%08" PFMT64x "  %10s %s\n", addr, "", m_intr);
+		} else {
+			char *str_buf = rz_str_newf("0x%08" PFMT64x "  %10s %s\n", addr, "", m_intr);
+			rz_strbuf_append(buf, str_buf);
+			free(str_buf);
+		}
+	}
+}
+
+static void disasm_print_ret(const RzCore *core, const RzOutputMode mode, const char *m_intr, const ut64 addr,
+	const bool ret_val, RzStrBuf *buf) {
+	switch (mode) {
+	case RZ_OUTPUT_MODE_QUIET:
+		if (!ret_val) {
+			rz_cons_printf("%s\n", m_intr);
+		} else {
+			rz_strbuf_append(buf, m_intr);
+		}
+		break;
+	default:
+		handle_default_disasm_print_mode(core, addr, ret_val, m_intr, buf);
+		break;
+	}
+}
+
+RZ_API RZ_BORROW int rz_core_disasm_until_ret(RzCore *core, ut64 addr, const int limit,
```

Comment:
`int` can't be *borrowed* or *owned*, only "objects", i.e. pointers in the C model. Also, is it really needs to be an API? I see it's not used outside of this file, could be just static function.

Reply:
Right, i wrongly mentioned it in the return value!, It should have been in mentioned for the buf pointer `buf`, will change it. Also this has been used in the `tui/rop.c`.